core/providers.tf

```
@@ -1,4 +1,11 @@
 terraform {
+  cloud {
+    organization = "nvminhtue"
+
+    workspaces {
+      tags = ["aws-infrastructure"]
+    }
+  }
```

What about moving this part out of `providers.tf` into a `main.tf` (same for shared)

```suggestion
  cloud {
    organization = "nvminhtue"

    workspaces {
      tags = ["aws-infrastructure"]
    }
  }
  
  # Terraform version
  required_version = "~> 1.6"
```

We can also add the required `terraform` version in the outer terraform block too (you need to create this block again in `main`). It seems you are using the latest (`1.6`) so I can't comment on any breaking changes or differences in the new versions though.

---

I don't see any benefit to the main file though, added the `required_version` in [5ee9924](https://github.com/nvminhtue/tyson_devops_ic/pull/20/commits/5ee9924687402849599d746fbe9911763778d325)

---

I actually don't know the real reasoning, seems we do it on projects though, ie

https://github.com/nimblehq/easyhotel-infra-aws/blob/develop/core/main.tf

---

I've seen some projects even put versions in a separate [versions.tf](https://github.com/terraform-aws-modules/terraform-aws-vpc/blob/bf9a89bf447a9c866dc0d30486aec5a24dbe2631/versions.tf) file 

or even some having a dedicated [terraform.tf ](https://github.com/trussworks/terraform-layout-example/blob/main/orgname-id/admin-global/terraform.tf)file for this 

Under the [AWS best practices](https://github.com/aws-samples/aws-terraform-best-practices#standard-module-structure) for terraform, they keep the terraform block and provider blocks under the `providers.tf`, so I'm also fine with this approach 👍